core/src/main/java/hudson/lifecycle/UnixLifecycle.java

```diff
@@ -98,10 +108,12 @@ public void verifyRestartable() throws RestartNotSupportedException {
         // the one described in http://www.nabble.com/Restarting-hudson-not-working-on-MacOS--to24641779.html
         //
         // according to http://www.mail-archive.com/wine-devel@winehq.org/msg66797.html this now works on Snow Leopard
-        if (Platform.isDarwin() && !Platform.isSnowLeopardOrLater())
-            throw new RestartNotSupportedException("Restart is not supported on Mac OS X");
-        if (args==null)
-            throw new RestartNotSupportedException("Failed to obtain the command line arguments of the process",failedToObtainArgs);
+        if (EXIT_ON_RESTART == null) {
```

Comment:
unrelated change?

Reply:
why? we do not need these validation if we use this option, right?

Reply:
sorry, NVM. You are right